imgui-glow-renderer/examples/02_triangle.rs

```diff
@@ -0,0 +1,82 @@
+//! A basic example showing imgui rendering together with some custom rendering.
+//!
+//! Note this example uses `RendererBuilder` rather than `auto_renderer` and
+//! (because we're using the default "trivial" `ContextStateManager`)
+//! therefore does not attempt to backup/restore OpenGL state.
+
+use std::time::Instant;
+
+use glow::HasContext;
+
+mod utils;
+
+use utils::Triangler;
+
+fn main() {
+    let (event_loop, window) = utils::create_window("Hello, triangle!", glutin::GlRequest::Latest);
+    let (mut winit_platform, mut imgui_context) = utils::imgui_init(&window);
+    let gl = utils::glow_context(&window);
+
+    let mut ig_renderer = imgui_glow_renderer::RendererBuilder::new()
+        .build_owning(gl, &mut imgui_context)
+        .expect("failed to create renderer");
+    let tri_renderer = Triangler::new(ig_renderer.gl_context(), "#version 330");
+
+    let mut last_frame = Instant::now();
+    event_loop.run(move |event, _, control_flow| {
+        *control_flow = glutin::event_loop::ControlFlow::Wait;
+        match event {
+            glutin::event::Event::NewEvents(_) => {
+                let now = Instant::now();
+                imgui_context
+                    .io_mut()
+                    .update_delta_time(now.duration_since(last_frame));
+                last_frame = now;
+            }
+            glutin::event::Event::MainEventsCleared => {
+                winit_platform
+                    .prepare_frame(imgui_context.io_mut(), window.window())
+                    .unwrap();
+
+                window.window().request_redraw();
+            }
+            glutin::event::Event::RedrawRequested(_) => {
+                {
+                    let gl = ig_renderer.gl_context();
+                    // This is required because, without the `StateBackupCsm`
+                    // (which is provided by `auto_renderer` but not
+                    // `RendererBuilder` by default), the OpenGL context is left
+                    // in an arbitrary, dirty state
+                    unsafe { gl.disable(glow::SCISSOR_TEST) };
+                    tri_renderer.render(gl);
+                }
+
+                let ui = imgui_context.frame();
+                // Safety: internally, this reference just gets passed as a
```

Comment:
This is completely incorrect. It's still UB to have a null reference.

Reply:
Fair. I'm assuming this means there's no way to do the equivalent of pass `NULL` here?

Reply:
Hmm, you can make it take an option and pass `None`, since `Option<&T>`/`Option<&mut T>` is guaranteed to have the same representation as a nullable pointer.

Reply:
Sorry, I haven't really looked to see if that answer makes sense. I'll be able to do a more thorough review later.